internal/arn/arn.go

```diff
@@ -60,7 +60,7 @@ type ARN struct {
 
 // Allows english letters, numbers, '.', '-', '_' and '/'. Starts with a
 // letter or digit. At least 1 character long.
-var validResourceIDRegex = regexp.MustCompile(`^[A-Za-z0-9][A-Za-z0-9_/\.-]*$`)
+var validResourceIDRegex = regexp.MustCompile(`[A-Za-z0-9_/\.-]+$`)
```

Comment:
quick question - why do we need this at all?

Reply:
Mainly because we don't otherwise restrict what characters can be in the resource ID - we generate the ARN in multiple places and may add more in the future. The caller can send any string, but we need it to be human readable at least and usable as a (Role ARN) query parameter.